### PR TITLE
chore(iac|integration): api-integration tests

### PR DESCRIPTION
**Pull request by rav3n11**

### CODE REVIEW

File: .github/workflows/deploy-auth.yml [hunk 1]
```diff
@@ -9,8 +9,6 @@ on:
         value: ${{ jobs.deploy.outputs.clientSecret }}
       userPoolId:
         value: ${{ jobs.deploy.outputs.userPoolId }}
-      userPoolClientId:
-        value: ${{ jobs.deploy.outputs.userPoolClientId }}
```
Comment (irumvanselme) [inline .github/workflows/deploy-auth.yml:13]: Why remove `userPoolClientId` from the outputs

Comment (irumvanselme) [inline .github/workflows/deploy-auth.yml:13]: Thanks for catching this `userPoolClientId ` is the same as `clientId`. Would you also remove the line in the iac/auth file.

Located at [this line](https://github.com/tabiya-tech/taxonomy-model-application/blob/d3374552a041fb4a23b2723cb6d2cf6dbed39742/iac/auth/src/index.ts#L154)

File: backend/test/integration/api.integration.test.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    }
+  });
+
+  describe("Test open routes", () => {
+    describe.each([...openRoutes])(`Test %s %s`, (route, method) => {
+      test(`should allow anonymous access to ${method} ${route}`, async () => {
+        // GIVEN a request to the open route with an anonymous token
+        const response = await generateRequest(request, route, method, `Bearer ${userTokens.anonymous}`).timeout(5000);
+        // THEN the response status code should be 200
+        expect(response.statusCode).toBe(StatusCodes.OK);
```
Comment (irumvanselme) [inline backend/test/integration/api.integration.test.ts:67]: may you use GIVEN, WHEN, THEN comments like.

// GIVEN a request with anonymous user

// WHEN the request is sent an open route ${route}

// THEN the response should be .....

File: backend/test/integration/api.integration.test.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      });
+
+      test(`should allow registered users access to ${method} ${route}`, async () => {
+        // GIVEN a request to the open route with a registered user token
+        const response = await generateRequest(
+          request,
+          route,
+          method,
+          `Bearer ${userTokens[AuthAPISpecs.Enums.TabiyaRoles.REGISTERED_USER]}`
+        ).timeout(5000);
```
Comment (irumvanselme) [inline backend/test/integration/api.integration.test.ts:77]: GIVEN, WHEN, THEN.

I think this would be the best way to go

**Review by irumvanselme — COMMENTED** [review]
Minor comments on tests.

Would you please use GIVEN, WHEN, THEN.

on the tests in file 
`backend/test/integration/api.integration.test.ts`

File: backend/test/integration/api.integration.test.ts (hunk 3 above)
Comment (rav3n11) [reply backend/test/integration/api.integration.test.ts:77]: Agreed. Changing....

**Pull request merged**